opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -2792,17 +2862,21 @@ namespace Opm {
             auto saved_previous_well_state = this->previous_well_state_;
             this->previous_well_state_ = this->well_state_;
 
-            well_container_ = createWellContainer(timeStepIdx);
-            for (auto& well : well_container_) {
-                well->init(&phase_usage_, depth_, gravity_, local_num_cells_, B_avg_);
-            }
+            this->well_container_ = this->createWellContainer(timeStepIdx);
```

Comment:
Is it necessary to recreate the whole container here?

Reply:
As of right now, it's not so much "recreate" as it is "create".  We (currently) run the enclosing function ([`runWellPIScaling`](https://github.com/OPM/opm-simulators/blob/0ff9e0c4bb861b93da73aae0053ebb4c34e9c147/opm/simulators/wells/BlackoilWellModel_impl.hpp#L2830)) as part of [`beginReportStep()`](https://github.com/OPM/opm-simulators/blob/0ff9e0c4bb861b93da73aae0053ebb4c34e9c147/opm/simulators/wells/BlackoilWellModel_impl.hpp#L296) at which point the `well_container_` does not yet have a well-defined value.

I didn't try too hard to call this function from `beginTimeStep()`&mdash;where `well_container_` **does** exist&mdash;because we need to guarantee that we don't scale connection transmissibility factors multiple times for the **same** `WELPI` keyword.